### ISSUE TITLE
[Shell Panel] Improve visibility of resizable functionality, allow `resizable` when `displayMode="float"`

### ISSUE THREAD

**Issue by macandcheese** [Esri/calcite-design-system#5652]
### Description

The current drag handle affordance can be a bit hard to notice. 
This should support both Shell Panel and upcoming Sheet implementations. 

### Acceptance Criteria

- [x] The resizable functionality of Shell Panel is easier to notice. This should also in certain cases be visually large enough (or have a hitbox large enough) to support thumb gestures for dragging.
- [x] Support `resizable` when the component is also `detached`, when `displayMode` is set to `"float"`

### Relevant Info
Questions:

- Should this have a "click to collapse" and "click to expand" button - brought up as UC feedback?

### Which Component

Shell Panel, Sheet (upcoming)

### Example Use Case

cc @asangma @SkyeSeitz @ashetland 

### Esri team

Calcite (design)

### Calcite Design Solution
<img width="1753" alt="image" src="https://github.com/user-attachments/assets/52adaa0f-7fdf-4935-9a68-89efe890971e">

🔗[ Link to Figma Specs]( https://www.figma.com/design/A7Z8XlYSkv5FSGwIJGoNu3/%5BShell-Panel%2C-Sheet%2C-Table%5D-%E2%80%9Cresizable%E2%80%9D-%5B7591%2C-5652%5D?node-id=1159-54157&m=dev&ready-for-dev=1)

**monday.com sync:** #5887227671

**Comment by macandcheese** [comment]
cc @SkyeSeitz @ashetland

**Comment by macandcheese** [comment]
+1 from BA cc @ddred11

**Comment by SkyeSeitz** [comment]
Final specs can be [found here](https://www.figma.com/file/A7Z8XlYSkv5FSGwIJGoNu3/%5BShell-Panel%2C-Sheet%2C-Table%5D-%E2%80%9Cresizable%E2%80%9D-%5B7591%2C-5652%5D?type=design&node-id=1%3A1435&mode=dev).

**Comment by COV-GIS** [comment]
I had been working my own footer component with resizing until it was brought to my attention a shell-panel works better and has resizing built in. At first I thought resizing wasn't working. I couldn't see the drag bar and when I did it was real hard to use on a large hi-res screen. And it isn't intuitive to the user there's resizing even if they notice the bar.

Overall I like how it works, looks and fits the "calcite design" well. It would fine by me if it was 1 or 2 px bigger, had an additional 5px of hitbox either side, and had some indication of draggability. Methinks having a brand colored portion of the bar in the center when not active would look good and clearly indicate draggability.

Here was my solution as an example.

![image](https://github.com/Esri/calcite-design-system/assets/14297903/25e9ddd7-abca-4ecb-9de7-953ed1127a52)

Also +1 on a collapse/expand button.

**Comment by driskull** [comment]
questions:

- There's no collapse button in the design right? I'm not seeing one. I just see double click to collapse/expand.

Some comments:

- We can maybe remove the need for the separator element to be focusable/focused for keyboard interactivity. Arrow keys when the panel is focused should just be able to handle this but would require an aria-live region to announce this.
- We can potentially use `interact.js` to handle resizing for better UX/consistency.
- I'm not sure the hit box area is really an issue. All other resizable areas including system windows don't really have a large hit box area for resizing. Even Figma's is pretty small.
- Neither Figma nor storybook has special styling for when a resize is active. Its basically just the focus styling. I'm not sure we need special styling either. If we make the shell-panel content container focusable it could handle both keyboard resizing and focus styles.

**Comment by macandcheese** [comment]
Plus 1 from Dev Summit

**Comment by driskull** [comment]
> Should this have a "click to collapse" and "click to expand" button - brought up as UC feedback?

I'm not sure a UI element for this is necessary.

We could explore double clicking to fully expand or collapse?

**Comment by driskull** [comment]
@macandcheese should we allow resizing for all display modes?

Currently, there is this restriction: `const allowResizing = displayMode !== "float-content" && displayMode !== "float" && resizable;
`

**Comment by macandcheese** [comment]
Yep - I think the naming of those values changed since this issue was open. It makes sense to me to allow it anywhere the dev needs.

**Comment by driskull** [comment]
@macandcheese 

There are some styles for max-width that exclude the float-all mode. So we may need a follow up issue to refactor these max-widths to be set for float-all mode as well.

https://github.com/Esri/calcite-design-system/blob/5ac406c73d57b47d08b6999a864bbe63255f2bd2/packages/calcite-components/src/components/shell-panel/shell-panel.scss#L57-L60

**Comment by DitwanP** [comment]
🍡 Verified locally on `dev`
